### PR TITLE
CIAC-10305 Fortinet Fortigate XDM Modeling Enhancement 

### PR DESCRIPTION
**Pull request by cweltPA**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-10305](https://jira-dc.paloaltonetworks.com/browse/CIAC-10305) , [CIAC-10409](https://jira-dc.paloaltonetworks.com/browse/CIAC-10409)

## Description
This PR enhances the existing XDM modeling for the Fortinet Fortigate content pack,
improving implementation of existing mappings, and adding support for multiple additional events and fields. 

### Mapping Updates

| CEF Log Field Name      | Fortigate Field Name | Field Description     | XDM Target Field| 
| :---        | :----       | :---          | :---       | 
|FTNTFGTappcat ~~FTNTFGTcat~~|appcat|Application Category|xdm.network.application_protocol_category|
|FTNTFGTapp|app|Application Name|xdm.target.application.name ~~xdm.source.application.name~~|
|FTNTFGTdevtype ~~FTNTFGTsrcfamily~~|devtype|Device type of the source|xdm.source.host.device_category|

### New Mappings Added

| CEF Log Field Name      | Fortigate Field Name | Field Description     | XDM Target Field| 
| :---        | :----       | :---          | :---       | 
|c6a2|c6a2|Source IPv6 Address|xdm.source.ipv6|
|c6a3|c6a3|Destination IPv6 Address|xdm.target.ipv6|
|deviceInboundInterface|srcintf|Interface name of the traffic's origin|xdm.source.interface|
|deviceOutboundInterface|dstintf|Interface of the traffic's destination|xdm.target.interface|
|dhost|dstname|Destination Hostname|xdm.target.host.hostname|
|dvchost|dvchost|Device Hostname|xdm.observer.name|
|externalID|sessionid|ID for the session|xdm.session_context_id, xdm.network.session_id|
|fname|filename|File name|xdm.target.file.filename|
|fsize|filesize|File size in bytes|xdm.target.file.size|
|FTNTFGTadgroup|adgroup|AD Group Name of FSSO user |xdm.source.user.groups|
|FTNTFGTap|ap|Access Point|xdm.intermediate.host.hostname|
|FTNTFGTapp|app|Application Name|xdm.target.application.name|
|FTNTFGTappcat|appcat|Application Category|xdm.network.application_protocol_category|
|FTNTFGTappid|appid|ID of the application. |xdm.target.application.name|
|FTNTFGTapplist|applist|Application Control profile name |xdm.network.rule|
|FTNTFGTapprisk|apprisk|Application Risk Level |xdm.alert.severity|
|FTNTFGTapsn|apsn|Access Point serial number |xdm.intermediate.host.hardware_uuid|
|FTNTFGTassigned|assigned|Assigned IP Address through PPPoE|xdm.source.host.ipv4_addresses, xdm.source.host.ipv6_addresses, xdm.source.host.ipv4_public_addresses|
|FTNTFGTassignip|assignip|IPsec VPN tunnel assigned IP address|xdm.network.vpn.allocated_ipv4, xdm.network.vpn.allocated_ipv6, xdm.intermediate.ipv4, xdm.intermediate.ipv6, xdm.intermediate.host.ipv4_addresses, xdm.intermediate.host.ipv4_public_addresses, xdm.intermediate.host.ipv6_addresses|
|FTNTFGTattack|attack|Attack Name|xdm.alert.original_threat_name, xdm.alert.risks|
|FTNTFGTattackid|attackid|Attack ID|xdm.alert.original_threat_id, xdm.alert.risks|
|FTNTFGTauditreporttype|auditreporttype|Security Rating report type|xdm.target.resource.type|
|FTNTFGTbanned_src|banned_src|NAC quarantine Banned Source IP |xdm.source.host.ipv4_addresses, xdm.source.host.ipv6_addresses, xdm.source.host.ipv4_public_addresses|
|FTNTFGTbssid|bssid|Basic service set ID of the WAP (Wireless Access Point)|xdm.intermediate.host.mac_addresses|
|FTNTFGTcatdesc|catdesc|FortiGuard web filter category Description|xdm.network.http.url_category|
|FTNTFGTccertissuer|ccertissuer|Client certificate issuer|xdm.network.tls.client_certificate.issuer|
|FTNTFGTcfgattr|cfgattr|Configuration attribute|xdm.target.resource.name|
|FTNTFGTcfgobj|cfgobj|Configuration object |xdm.target.resource.value|
|FTNTFGTcfgpath|cfgpath|Configuration path|xdm.target.resource.type|
|FTNTFGTcipher|cipher|Encryption Type|xdm.network.tls.cipher|
|FTNTFGTcommand|command|Shell command|xdm.target.process.command_line|
|FTNTFGTcveid|cveid|CVE ID|xdm.alert.risks|
|FTNTFGTdaddr|daddr|Destination address |xdm.target.host.ipv4_addresses, xdm.target.host.ipv6_addresses, xdm.target.host.ipv4_public_addresses|
|FTNTFGTdevintfname|devintfname|HA device interface name |xdm.target.interface|
|FTNTFGTdevtype|devtype|Device type of the source|xdm.source.host.device_category|
|FTNTFGTdhcp_msg|dhcp_msg|DHCP Message|xdm.network.dhcp.message_type|
|FTNTFGTdst_host|dst_host|Destination Host|xdm.target.host.hostname|
|FTNTFGTdstauthserver|dstauthserver|Destination Auth Server|xdm.target.host.hostname|
|FTNTFGTdstcity|dstcity|Destination host city|xdm.target.location.city|
|FTNTFGTdstcountry|dstcountry|Destination country|xdm.target.location.country|
|FTNTFGTdstdevtype|dstdevtype|Destination Device Type |xdm.target.host.device_category|
|FTNTFGTdstfamily|dstfamily|Destination Host OS Platform Family|xdm.target.host.device_category|
|FTNTFGTdstfamily|dstfamily|Destination OS family|xdm.target.host.os_family|
|FTNTFGTdsthwversion|dsthwversion|Destination hardware version|xdm.target.host.device_model|
|FTNTFGTdstosname|dstosname|Destination OS name|xdm.target.host.os|
|FTNTFGTdstregion|dstregion|Destination Host Region|xdm.target.location.region|
|FTNTFGTdstuser|dstuser|Destination user|xdm.target.user.username|
|FTNTFGTerror_num|error_num|Error Number |xdm.event.outcome_reason|
|FTNTFGTerror|error|DNS filter service error message|xdm.event.outcome_reason|
|FTNTFGTeventtype|eventtype|Event Type|xdm.network.dns.is_response|
|FTNTFGTfctuid|fctuid|FortiClient User ID |xdm.source.user.identifier|
|FTNTFGTfilehash|filehash|Used by Outbreak Prevention External Hash: the hash signature used in the detection|xdm.target.file.sha256, xdm.target.file.md5|
|FTNTFGTfiletype|filetype|File type|xdm.target.file.file_type |
|FTNTFGTforwardedfor|forwardedfor|X-Forwarded-For HTTP header |xdm.network.http.http_header.header, xdm.network.http.http_header.value, xdm.source.host.ipv4_addresses, xdm.source.host.ipv4_public_addresses, xdm.source.host.ipv6_addresses|
|FTNTFGTgateway|gateway|Gateway ip address for PPPoE status report |xdm.intermediate.ipv4, xdm.intermediate.ipv6, xdm.intermediate.host.ipv4_addresses, xdm.intermediate.host.ipv4_public_addresses, xdm.intermediate.host.ipv6_addresses|
|FTNTFGTgroup|group|User Group Name|xdm.source.user.groups|
|FTNTFGThttpcode|httpcode|HTTP Code|xdm.network.http.response_code|
|FTNTFGThttpmethod|httpmethod|HTTP request method|xdm.network.http.method|
|FTNTFGTicmpcode|icmpcode|ICMP code|xdm.network.icmp.code|
|FTNTFGTicmptype|icmptype|The type of ICMP message|xdm.network.icmp.type|
|FTNTFGTincidentserialno|incidentserialno|Incident serial number |xdm.alert.original_alert_id, xdm.alert.risks|
|FTNTFGTinitiator|initiator|The initiator user for override|xdm.source.user.username|
|FTNTFGTinvalidmac|invalidmac|Detected MAC address with invalid OUI|xdm.target.host.mac_addresses|
|FTNTFGTipaddr|ipaddr|IP addresses from DNS response answer section|xdm.network.dns.dns_resource_record.value|
|FTNTFGTlease|lease|DHCP lease time |xdm.network.dhcp.lease|
|FTNTFGTlocal|local|Local IP for a PPPD Connection|xdm.source.host.ipv4_addresses, xdm.source.host.ipv6_addresses, xdm.source.host.ipv4_public_addresses|
|FTNTFGTlogin|login|SSH login Name |xdm.source.user.username|
|FTNTFGTmanuf|manuf|Manufacturer name|xdm.source.host.manufacturer|
|FTNTFGTmasterdstmac|masterdstmac|Destination master MAC address |xdm.target.host.mac_addresses|
|FTNTFGTmastersrcmac|mastersrcmac|The master MAC address for a host that has multiple network interfaces|xdm.source.host.mac_addresses|
|FTNTFGTnat|nat|NAT IP Address|xdm.source.ipv4, xdm.source.ipv6, xdm.source.host.ipv4_addresses, xdm.source.host.ipv6_addresses|
|FTNTFGTnew_status|new_status|New status|xdm.target.resource.value|
|FTNTFGTnew_value|new_value|New value|xdm.target.resource.value|
|FTNTFGTnewvalue|newvalue|New value|xdm.target.resource.value|
|FTNTFGTold_status|old_status|Original Status|xdm.target.resource_before.value|
|FTNTFGTold_value|old_value|Original value|xdm.target.resource_before.value|
|FTNTFGToldvalue|oldvalue|Original value|xdm.target.resource_before.value|
|FTNTFGToldwprof|oldwprof|Old Web Filter Profile|xdm.target.resource_before.value|
|FTNTFGTopercountry|opercountry|The name of operating country on a AP|xdm.intermediate.location.country|
|FTNTFGTpolicy_id|policy_id|Name of the firewall policy governing the traffic which caused the log message|xdm.network.rule|
|FTNTFGTpolicyid|policyid|Name of the firewall policy governing the traffic which caused the log message|xdm.network.rule|
|FTNTFGTpoolname|poolname|IP Pool Name |xdm.target.resource.name|
|FTNTFGTprofile|profile|Profile Name|xdm.network.rule|
|FTNTFGTqclass|qclass|DNS Query class|xdm.network.dns.dns_question.class, xdm.network.dns.dns_resource_record.class|
|FTNTFGTqname|qname|Query domain name|xdm.network.dns.dns_question.name, xdm.network.dns.dns_resource_record.name|
|FTNTFGTqtype|qtype|Query type description|xdm.network.dns.dns_question.type, xdm.network.dns.dns_resource_record.type|
|FTNTFGTreferralurl|referralurl|Referrer URI|xdm.network.http.referrer|
|FTNTFGTremote|remote|IP Address of the PPP Remote end |xdm.target.ipv4, xdm.target.ipv6, xdm.target.host.ipv4_addresses, xdm.target.host.ipv4_public_addresses, xdm.target.host.ipv6_addresses|
|FTNTFGTreporttype|reporttype|Report Type|xdm.target.resource.type|
|FTNTFGTsaddr|saddr|Source Address IP|xdm.source.host.ipv4_addresses|
|FTNTFGTserialno|serialno|Serial Number|xdm.target.host.hardware_uuid|
|FTNTFGTserver|server|Server IP Address|xdm.target.host.ipv4_addresses, xdm.target.host.ipv6_addresses, xdm.target.host.ipv4_public_addresses|
|FTNTFGTshapingpolicyid|shapingpolicyid|Shaping Policy ID |xdm.network.rule|
|FTNTFGTsn|sn|Serial Number|xdm.target.host.hardware_uuid|
|FTNTFGTsnclosest|snclosest|SN of the AP closest to the rogue AP |xdm.intermediate.host.hardware_uuid|
|FTNTFGTsrccity|srccity|Source host city|xdm.source.location.city|
|FTNTFGTsrccountry|srccountry|Source host country|xdm.source.location.country|
|FTNTFGTsrchwvendor|srchwvendor|Source hardware vendor|xdm.source.host.manufacturer|
|FTNTFGTsrchwversion|srchwversion|Source hardware version|xdm.source.host.device_model|
|FTNTFGTsrcregion|srcregion|Source Host Region|xdm.source.location.region|
|FTNTFGTsrcuuid|srcuuid|UUID of the Source Address Object|xdm.source.host.device_id|
|FTNTFGTsslaction|sslaction|Action taken by ssl-ssh-profile|xdm.observer.action|
|FTNTFGTstamac|stamac|The MAC address of wifi station|xdm.intermediate.host.mac_addresses|
|FTNTFGTtamac|tamac|The MAC address of Transmitter. If none, then Receiver |xdm.source.host.mac_addresses|
|FTNTFGTthreattype|threattype|WIDS threat type |xdm.alert.original_threat_name, xdm.alert.risks|
|FTNTFGTtlsver|tlsver|TLS version|xdm.network.tls.protocol_version|
|FTNTFGTtrueclntip|trueclntip|True-Client-IP|xdm.source.host.ipv4_addresses, xdm.source.host.ipv6_addresses, xdm.source.host.ipv4_public_addresses|
|FTNTFGTtunnelid|tunnelid|IPsec VPN tunnel ID|xdm.intermediate.host.device_id|
|FTNTFGTtunnelip|tunnelip|IPsec VPN tunnel IP address|xdm.network.vpn.allocated_ipv4, xdm.network.vpn.allocated_ipv6, xdm.intermediate.ipv4, xdm.intermediate.ipv6, xdm.intermediate.host.ipv4_addresses, xdm.intermediate.host.ipv4_public_addresses, xdm.intermediate.host.ipv6_addresses|
|FTNTFGTutmaction|utmaction|Security action performed by UTM|xdm.observer.action|
|FTNTFGTvirus|virus|Virus Name|xdm.alert.original_threat_name, xdm.alert.risks|
|FTNTFGTviruscat|viruscat|Virus Category|xdm.alert.category, xdm.alert.risks|
|FTNTFGTvirusid|virusid|Virus ID (unique virus identifier)|xdm.alert.original_threat_id, xdm.alert.risks|
|FTNTFGTvoip_proto|voip_proto|VOIP protocol: SIP/SCCP/MGCP/h323|xdm.network.application_protocol|
|FTNTFGTvpntunnel|vpntunnel|IPsec VPN Tunnel Name|xdm.intermediate.host.hostname|
|FTNTFGTvulncat|vulncat|Vulnerability Category|xdm.alert.risks|
|FTNTFGTvulnid|vulnid|Vulnerability ID|xdm.alert.risks|
|FTNTFGTvulnname|vulnname|Vulnerability name|xdm.alert.risks|
|FTNTFGTxauthgroup|xauthgroup|IPsec VPN Xauth user group name|xdm.intermediate.user.groups|
|FTNTFGTxauthuser|xauthuser|IPsec VPN Xauth user name |xdm.intermediate.user.username|
|request|url|Target URL|xdm.target.url|
|requestClientApplication|agent|User Agent|xdm.source.user_agent|
|requestContext|catdesc|DNS category description|xdm.network.http.url_category|
|shost|shost|Source Hostname| xdm.source.host.hostname|
|sourceTranslatedAddress|transip|NAT source IP|xdm.source.host.ipv4_addresses, xdm.source.host.ipv6_addresses, xdm.source.host.ipv4_public_addresses|
|sourceTranslatedPort|transport|NAT source port|xdm.source.port|

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by OBavly — APPROVED** [review]

File: Packs/FortiGate/ReleaseNotes/2_0_2.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+
+#### Modeling Rules
+
+##### Fortinet FortiGate Modeling Rule
+
+- Updated XDM mappings for the following fields: 
+| CEF Log Field Name      | Fortigate Field Name | Field Description     | XDM Target Field| 
+| :---        | :----       | :---          | :---       | 
+`FTNTFGTapp`|app|Application Name|`xdm.target.application.name` ~~`xdm.source.application.name`~~|
```
Comment (ShirleyDenkberg) [inline Packs/FortiGate/ReleaseNotes/2_0_2.md:9]: Is all this information really needed in the release notes?

File: Packs/FortiGate/ReleaseNotes/2_0_2.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+`FTNTFGTappcat` ~~FTNTFGTcat~~|appcat|Application Category|`xdm.network.application_protocol_category`|
+`FTNTFGTdevtype` ~~FTNTFGTsrcfamily~~|devtype|Device type of the source|`xdm.source.host.device_category`|
+
+- Added XDM mappings for the following fields: 
+| CEF Log Field Name      | Fortigate Field Name | Field Description     | XDM Target Field| 
+| :---        | :----       | :---          | :---       | 
+`c6a2`|c6a2|Source IPv6 Address|`xdm.source.ipv6`|
```
Comment (ShirleyDenkberg) [inline Packs/FortiGate/ReleaseNotes/2_0_2.md:16]: Here too.  Do we really need all this information in the release notes?

**Comment by ShirleyDenkberg** [comment]
@OBavly Doc review completed.
@cweltPA Do we really need so much info in the Release Notes?  It's a bit overwhelming and difficult to read.

File: Packs/FortiGate/ReleaseNotes/2_0_2.md (hunk 1 above)
Comment (cweltPA) [reply Packs/FortiGate/ReleaseNotes/2_0_2.md]: @ShirleyDenkberg Thanks for the quick feedback. 
I updated the release notes to be more concise. Can you please re-review ?

File: Packs/FortiGate/ReleaseNotes/2_0_2.md (hunk 2 above)
Comment (cweltPA) [reply Packs/FortiGate/ReleaseNotes/2_0_2.md]: Updated to be more concise.

File: Packs/FortiGate/ReleaseNotes/2_0_2.md (hunk 1 above)
Comment (ShirleyDenkberg) [reply Packs/FortiGate/ReleaseNotes/2_0_2.md]: Much better.  Thanks.

**Pull request merged**